### ISSUE TITLE
Warn about using bash-only variables on a declared /bin/sh

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2093]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

echo "${BASH_SOURCE[0]}:${FUNCNAME[0]}:$LINENO $BASH_COMMAND"

```

#### Here's what shellcheck currently says:

(shellcheck.net is down, I cannot add the existing messages)

#### Here's what I wanted or expected to see:

* In POSIX sh, BASH_SOURCE, FUNCNAME, BASH_COMMAND are not defined. (Bashism)

**Comment by emanuele6** [comment]
> (shellcheck.net is down, I cannot add the existing messages)

```

Line 3:
echo "${BASH_SOURCE[0]}:${FUNCNAME[0]}:$LINENO $BASH_COMMAND"
      ^-- SC3054: In POSIX sh, array references are undefined.
                        ^-- SC3054: In POSIX sh, array references are undefined.

```

---------------------

But for this script:
```sh
#!/bin/sh

echo "$BASH_COMMAND"
```
The output of `shellcheck` is:
```
 No issues detected!

```

**Comment by koalaman** [comment]
A check has been added for a bunch of these bash specific variables. Thanks!
